### PR TITLE
CCSMCDS-115/116 Fix two Toggle issue

### PR DESCRIPTION
**Pull request by yunwwang**
This PR fixes two issues related to Toggle buttons

* Change code and valueCodeableConcept of Immuno Suppressed observation to match CQL library
* Preseve the original pregnant status from Patient's FHIR data

### CODE REVIEW

File: src/hooks/useCds/translate.js [hunk 1]
```diff
@@ -259,6 +259,15 @@ const loincBiopsyReport = {
 const immunosuppressedObservation = {
   label: 'Immunosuppressed',
   code: {
+    coding: [
+      {
+        system: LOINC_URL,
+        code: 96381-9,
```
Comment (mickohanlon23) [inline src/hooks/useCds/translate.js]: Does new immunosuppressed LOINC code need single quotes, like the other codes we define here?

Comment (mickohanlon23) [reply src/hooks/useCds/translate.js]: Upon testing locally with https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/136 as a dependency, the wrong code is passed onto the patient as-is, and toggling the immunocompromised switch does not cause a change in patient data. 

When this code is placed in single quotes, the correct code is passed onto the patient, and you can see one of the Logica Health patients getting a screening recommendation from the Immunocompromised logic path as opposed to the average risk logic path.

File: src/hooks/useCds/index.js [hunk 2]
```diff
@@ -11,27 +11,32 @@ import { stridesData } from './strides';
...
  */
-export const useCds = (patientData, toggleStatus, isToggleChanged) => {
+export const useCds = (patientData, toggleStatus) => {
 
   const [output, setOutput] = useState({});
   const [isLoadingCdsData, setIsLoadingCdsData] = useState(false);
+  const [isPregnant, setIsPreganant] = useState(false);
 
   useEffect(() => {
+    if (patientData.length == 0) {
```
Comment (mickohanlon23) [inline src/hooks/useCds/index.js]: eslint
```suggestion
    if (patientData.length === 0) {
```

File: src/smart/SmartPatient.js [hunk 3]
```diff
@@ -15,24 +15,13 @@ export function SmartPatient() {
     isImmunosuppressed: false,
     isPregnant: false,
     isPregnantConcerned: false,
-    isSymptomatic: false
+    isSymptomatic: false,
+    isToggleChanged: false
```
Comment (mickohanlon23) [inline src/smart/SmartPatient.js:18]: Can the `istoggleChanged` useState declaration on line 13 be deleted now?

```javascript
  const [isToggleChanged, setIsToggleChanged] = useState(false);
```

File: src/smart/SmartPatient.js [hunk 4]
```diff
@@ -15,24 +15,13 @@ export function SmartPatient() {
...
   });
 
-  const { output: dashboardInput, isLoadingCdsData: isLoadingCdsData } = useCds(patientData, toggleStatus, isToggleChanged);
+  const { output: dashboardInput, isLoadingCdsData: isLoadingCdsData } = useCds(patientData, toggleStatus);
```
Comment (mickohanlon23) [inline src/smart/SmartPatient.js]: eslint is now throwing the following warning for the `isLoadingCdsData` declaration here:

> Destructuring assignment isLoadingCdsData unnecessarily renamed.

I'm not sure why this wasn't being recognized before, but should we consider switching to:
```javascript
  const { output: dashboardInput, isLoadingCdsData } = useCds(patientData, toggleStatus);

```

**Review by mickohanlon23 — CHANGES_REQUESTED** [review]
## Overview
Awesome! Some minor changes requested, but overall this accomplishes the two things we wanted to accomplish:

## Tests
### Immunocompromised Toggle
- I tested by changing the `ccsm-cds-with-tests` dependency to a local dependency using the branch in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/136. Using the Logica sandbox, I tested using any patient, since the general, "Cervical Cancer Screening is Due" recommendation from the average risk screening logic path should switch to a screening recommendation from the Immunocompromised logic path when the immmunocompromised toggle switch is toggled for a patient with no relevant cervical cancer screening and management history. When the new immunocompromised code in `translate.js` is put in single quotes, then this behavior works as expected!
### Pregnancy Toggle
- I tested with the following Logica Sandbox patient: `MITRE AGC-Cytology`. Here I wanted to test that a patient who normally gets a recommendation that includes an option for treatment would get a different recommendation when they are deemed pregnant by the toggle switch. This indeed was the case with this patient. And as was changed in this PR, the patient now starts with the "PREGNANT" status field at "No", and the field doesn't change when "Pregnant" is toggled on.
   - Note: flipping the "Pregnant" toggle switch for this patient revealed a bug in that the recommendation gets stuck at "Loading Recommendation", however, this appears to be unrelated to this PR. I confirmed that the behavior still existed before this PR, and it seems to be because there is no recommendation sent for a patient with AGC who is pregnant, but also, this scenario does not produce an error.
- I tested with the following Logica sandbox patient: `MITRE Pregnant`. This was to show that a patient who is already pregnant would not have their "PREGNANT" status affected by toggling the "Pregnant" toggle. This patient did correctly start with "PREGNANT: Yes" in their display, and then that display remained after I toggled the toggle switch.

## Conclusion
Once the few comments I made are cleaned up, this should be good to merge. Note that we will want to merge this after https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/136, and then bump the version of ccsm-cds-with-tests that the dashboard depends on so that we can leverage the new immunocompromised Observation logic.

**Comment by yunwwang** [comment]
Comments adopted and fixed

**Review by mickohanlon23 — APPROVED** [review]
All Review comments were addressed. Thanks for revising this so quickly, @yunwwang! Once https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/136 is merged, we should update the ccsm-cds-with-tests dependency and then merge this in.

**Pull request merged**